From 12b82d279f9dfdf0c88ce9b2758cebe4b3044f47 Mon Sep 17 00:00:00 2001
From: Abhishek Chaudhary <quic_ach@quicinc.com>
Date: Mon, 8 Aug 2022 18:18:55 +0530
Subject: [PATCH 089/500] Add support for first IP fragment dissection

Add support for dissecting ports from first fragment of the
IP packet. This is needed for PPE to find the 5-tuple/flow
to flush the rule based on exception due to fragmented
IP packets.

Change-Id: Ibb372e60381ed232dbaecd5d63a7d2fa6ae2dd9a
Signed-off-by: Abhishek Chaudhary <quic_ach@quicinc.com>
---
 net/core/flow_dissector.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/core/flow_dissector.c b/net/core/flow_dissector.c
index 272f09251343..540fc531cac3 100644
--- a/net/core/flow_dissector.c
+++ b/net/core/flow_dissector.c
@@ -1627,9 +1627,10 @@ bool __skb_flow_dissect(const struct net *net,
 		break;
 	}
 
-	if (!(key_control->flags & FLOW_DIS_IS_FRAGMENT))
+	if (!(key_control->flags & FLOW_DIS_IS_FRAGMENT) || (key_control->flags & FLOW_DIS_FIRST_FRAG)) {
 		__skb_flow_dissect_ports(skb, flow_dissector, target_container,
 					 data, nhoff, ip_proto, hlen);
+	}
 
 	/* Process result of IP proto processing */
 	switch (fdret) {
-- 
2.34.1

